datadog/api/api_client.py

```diff
@@ -0,0 +1,213 @@
+# stdlib
+import logging
+import time
+
+# datadog
+from datadog.api import _api_version, _max_timeouts, _backoff_period
+from datadog.api.exceptions import ClientError, ApiError, HttpBackoff, \
+    HttpTimeout, ApiNotInitialized
+from datadog.api.http_client import RequestClient
+from datadog.util.compat import json, is_p3k
+
+
+log = logging.getLogger('dd.datadogpy')
+
+
+class APIClient(object):
+    """
+    """
+    # HTTP transport parameters
+    _backoff_period = _backoff_period
+    _max_timeouts = _max_timeouts
+    _backoff_timestamp = None
+    _timeout_counter = 0
+    _api_version = _api_version
+
+    def __init__(self):
+        """
+        Instantiate a HTTP Client.
+        """
+        self._http_client = self._get_http_client()
+
+    @staticmethod
+    def _get_http_client():
+        """
+        Resolve the appropriate HTTP client based on priority and user environment.
+        """
+
+        return RequestClient
+
+    def submit(self, method, path, body=None, attach_host_name=False, response_formatter=None,
```

Comment:
Related to the previous comment, this method should be a class method: it only uses class methods/variables.